h5pt.go

```diff
@@ -81,40 +93,170 @@ func (t *Table) ReadPackets(start, nrecords int, data interface{}) error {
 	return h5err(err)
 }
 
+func extractStructValues(rv reflect.Value, rt reflect.Type) (ptr unsafe.Pointer, err error) {
+	// Initially the memory will be allocated to 64 bytes. If more is required
+	// later, this will be increased.
+	memSize := C.size_t(64)
+	ptr = C.malloc(memSize)
+
+	for i := 0; i < rv.NumField(); i++ {
+		var dataPtr unsafe.Pointer
+		var dataSize C.size_t
+		f := rv.Field(i)
+		ft := string(f.Type().Name())
+		offset := C.size_t(rt.Field(i).Offset)
+
+		switch ft {
```

Comment:
perhaps use `ft := f.Type().Kind()` and switch on `reflect.Int8`, `reflect.Uint8`, ... instead of switching on and comparing strings ?

Reply:
No problem :) Fixed